src/components/button-bar/button-bar.pw.tsx

```diff
@@ -0,0 +1,239 @@
+
+  await buttonAtIndex(0).focus();
+  await buttonAtIndex(0).press("Tab");
+  await expect(buttonAtIndex(1)).toBeFocused();
+  await expect(buttonAtIndex(0)).not.toBeFocused();
+
+  await buttonAtIndex(1).press("Tab");
+  await expect(buttonAtIndex(2)).toBeFocused();
+  await expect(buttonAtIndex(1)).not.toBeFocused();
+});
+
+test.describe(
+  "when custom Button wrapper components are used as children in ButtonBar",
+  async () => {
+    test("Button size is small when the size prop is set to small and passed to ButtonBar", async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ButtonBarWithWrapper size="small" />);
+
+      for (let i = 0; i < 3; i++) {
+        // eslint-disable-next-line no-await-in-loop
+        const buttonWidth = await getStyle(
+          buttonDataComponent(page).nth(i),
+          "width"
+        );
+        // eslint-disable-next-line no-await-in-loop
+        await expect(parseFloat(buttonWidth)).toBeCloseTo(82, 0);
+      }
+    });
+
+    [
+      { iconPosition: "before" as const, expectedMargin: "margin-right" },
+      { iconPosition: "after" as const, expectedMargin: "margin-left" },
+    ].forEach(({ iconPosition, expectedMargin }) => {
+      test(`Button Icon position is ${iconPosition} text when the iconPosition is set and passed to ButtonBar`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<ButtonBarWithWrapper iconPosition={iconPosition} />);
+        await expect(icon(page).nth(0)).toHaveCSS(expectedMargin, "8px");
+        await expect(icon(page).nth(1)).toHaveCSS(expectedMargin, "8px");
+        await expect(icon(page).nth(2)).toHaveCSS(expectedMargin, "8px");
+      });
+    });
+
+    test("should verify ButtonBar with wrapped components can be navigated using keyboard", async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ButtonBarWithWrapper />);
+
+      const buttonAtIndex = (index: number) =>
+        buttonDataComponent(page).nth(index);
+
+      await buttonAtIndex(0).focus();
+      await buttonAtIndex(0).press("Tab");
+      await expect(buttonAtIndex(1)).toBeFocused();
+      await expect(buttonAtIndex(0)).not.toBeFocused();
+
+      await buttonAtIndex(1).press("Tab");
+      await expect(buttonAtIndex(2)).toBeFocused();
+      await expect(buttonAtIndex(1)).not.toBeFocused();
+
+      await buttonAtIndex(2).press("Tab");
+
+      await expect(page.getByLabel("csv")).toBeFocused();
+      await expect(buttonAtIndex(2)).not.toBeFocused();
+    });
+  }
+);
+
+test.describe("renders with ButtonMinor children", async () => {
+  const indexes = [0, 1, 2];
+
+  test.beforeEach(async ({ mount }) => {
+    await mount(<ButtonBarMinor />);
+  });
+
+  indexes.forEach((index) => {
+    test(`should apply correct background-color on hover for ${index} ButtonMinor children`, async ({
+      page,
+    }) => {
+      const minorButton = buttonMinorComponent(page, index);
```

Comment:
Similarly, can this const be set at the top of the file as it's repeated in other test cases? Or does it need to be repeated in each test case.

Reply:
nope, it should be re created in each test to have the `page` context